File: JavaScript solutions/121. Best Time to Buy and Sell Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  let n = prices.length;
  if (n <= 1) {
    return 0;
  }
  let min_price = prices[0];
  let max_profit = 0;
  for (let i = 1; i < n; i++) {
    max_profit = Math.max(max_profit, prices[i] - min_price);
    min_price = Math.min(min_price, prices[i]);
  }
  return max_profit;
};

File: C++ solutions/121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        if (n <= 1) {
            return 0;
        }
        int min_price = prices[0];
        int max_profit = 0;
        for (int i = 1; i < n; i++) {
            max_profit = max(max_profit, prices[i] - min_price); 
            min_price = min(min_price, prices[i]);
        }
        return max_profit; 
    }
};